src/Content/Text/HTML.php

```diff
@@ -961,4 +961,61 @@ public static function unamp($s)
 	{
 		return str_replace('&amp;', '&', $s);
 	}
+
+	/**
+	 * Clean an HTML text for potentially harmful code
+	 *
+	 * @param string $text
+	 * @param array  $allowedIframeDomains List of allowed iframe source domains without the scheme
+	 * @return string
+	 */
+	public static function purify(string $text, array $allowedIframeDomains = []): string
+	{
+		// Allows cid: URL scheme
+		\HTMLPurifier_URISchemeRegistry::instance()->register('cid', new HTMLPurifier_URIScheme_cid());
+
+		$config = \HTMLPurifier_HTML5Config::createDefault();
+		$config->set('HTML.Doctype', 'HTML5');
+
+		// Used to remove iframe with src attribute filtered out
+		$config->set('AutoFormat.RemoveEmpty', true);
+
+		$config->set('HTML.SafeIframe', true);
+
+		array_walk($allowedIframeDomains, function (&$domain) {
+			// Allow the domain and all its eventual sub-domains
+			$domain = 'https://(?:(?!-)[A-Za-z0-9-]{1,63}(?<!-)\.)*' . preg_quote($domain, '%');
```

Comment:
Wouldn't this regular expression also be valid when you enter `youtube.com` but the URL points to `https://fakeyoutube.com` or `https://youtube.com.fake.tld`?

Reply:
The first wouldn't be allowed, but the second would, I have to add a trailing `/` to the expression.

Reply:
This has been addressed.